.changeset/make-formik-possibly-synchronous.md

```diff
@@ -0,0 +1,5 @@
+---
+'formik': minor
```

Comment:
@quantizor we might want to make this a major version bumb in the end.

I had not thought of that initially but if anyone implemented things as `setFieldValue(..).catch(...)` then it would break if their validation functions would be sync. It would be completely fine if they would implement it as `await setFieldValue(..)` or `void setFieldValue(..)` though.

What are your thoughts?